### PR TITLE
add seq_db to enable phmmer searches

### PR DESCRIPTION
**Pull request by FelixLanger**
I added I sequence_database class with a phmmer method to allow a phmmer search against that database.
The binaries for phmmer still need to be added and should probably all be updated to the new hmmer version 3.3.2

### CODE REVIEW

File: hmmer/hmmer.py [hunk 1]
```diff
@@ -268,3 +267,45 @@ def _match(self, bin: Path, target: Path, options: Options) -> Result:
             check_call(cmd_match)
 
         return Result(options.tblout, options.domtblout)
+
+
+class seq_db:
```
Comment (horta) [inline hmmer/hmmer.py:272]: Shall we have it named as `SeqDB` instead? It is often the case in Python that we adopt upper-case of the first letter and then Camel case for class names. (There are some exceptions like in numpy, for example, but the non-core libraries tend to following that camel case for class name as far as I know).

**Comment by horta** [comment]
Apart from the class name it looks all good to me. I'm happy to merge after that modification =)

**Comment by FelixLanger** [comment]
sure, I changed the name

**Comment by horta** [comment]
Thanks a lot! I will review it and make a new release today!

**Pull request merged**